### PR TITLE
[PW-2768] Payment Status API endpoint

### PR DESCRIPTION
**Pull request by acampos1916**
## Summary
Created a new Payment Status endpoint and service to return the action and payment response result code.

## Tested scenarios
NA

**Fixed issue**:  PW-2768

### CODE REVIEW

File: src/Controller/SalesChannelApiController.php [hunk 1]
```diff
@@ -144,14 +153,15 @@ public function postPaymentDetails(
...
      *
+     * @param Request $request
      * @param SalesChannelContext $context
      * @return JsonResponse
      */
-    public function getPaymentStatus(SalesChannelContext $context): JsonResponse
+    public function getPaymentStatus(Request $request): JsonResponse
     {
-        return new JsonResponse(['aloha']);
+        return new JsonResponse($this->paymentStatusService->getPaymentStatusWithOrderId($request->get('orderId')));
```
Comment (cyattilakiss) [inline src/Controller/SalesChannelApiController.php:165]: you could still use the function I created with the SalesChannelContext token and the order id - is there a reason why you only filter by order id?

File: src/Service/PaymentStatusService.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    public function __construct(
+        PaymentResponseService $paymentResponseService
+    ) {
+        $this->paymentResponseService = $paymentResponseService;
+    }
+
+    public function getPaymentStatusWithOrderId(string $orderId): array
+    {
+        $paymentResponse = $this->paymentResponseService->getWithOrderId($orderId);
+        $responseData = json_decode($paymentResponse->getResponse(), true);
```
Comment (cyattilakiss) [inline src/Service/PaymentStatusService.php:44]: Adding some validation 
```suggestion
        $responseData = json_decode($paymentResponse->getResponse(), true);
        
        if (json_last_error() !== JSON_ERROR_NONE) {
            //TODO error handling
        }
```

File: src/Service/PaymentStatusService.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        $action = [];
+
+        if ($responseData['action']) {
+            $action = [$responseData['action']];
+        }
+
+        $paymentStatus = [
+            'paymentResponseResultCode' => $paymentResponse->getResultCode(),
```
Comment (cyattilakiss) [inline src/Service/PaymentStatusService.php:52]: To keep it consistent
```suggestion
            'resultCode' => $paymentResponse->getResultCode(),
```

File: src/Service/PaymentStatusService.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        $paymentResponse = $this->paymentResponseService->getWithOrderId($orderId);
+        $responseData = json_decode($paymentResponse->getResponse(), true);
+        $action = [];
+
+        if ($responseData['action']) {
+            $action = [$responseData['action']];
+        }
+
+        $paymentStatus = [
```
Comment (cyattilakiss) [inline src/Service/PaymentStatusService.php:51]: what do you think about adding isFinal into this one as well, is there any benefit of it?
https://github.com/Adyen/adyen-shopware6/blob/c4b34fe5b622e5f5e9e6b6b4025b9d14def6edeb/src/Handlers/PaymentResponseHandler.php#L257

Comment (cyattilakiss) [reply src/Service/PaymentStatusService.php:51]: Actually you can simply just use this function: https://github.com/Adyen/adyen-shopware6/blob/c4b34fe5b622e5f5e9e6b6b4025b9d14def6edeb/src/Handlers/PaymentResponseHandler.php#L247

Comment (acampos1916) [reply src/Service/PaymentStatusService.php]: Done!

File: src/Controller/SalesChannelApiController.php (hunk 1 above)
Comment (acampos1916) [reply src/Controller/SalesChannelApiController.php]: Done!

File: src/Controller/SalesChannelApiController.php [hunk 5]
```diff
@@ -144,14 +153,20 @@ public function postPaymentDetails(
...
      * @param SalesChannelContext $context
      * @return JsonResponse
      */
-    public function getPaymentStatus(SalesChannelContext $context): JsonResponse
+    public function getPaymentStatus(Request $request, SalesChannelContext $context): JsonResponse
     {
-        return new JsonResponse(['aloha']);
+        return new JsonResponse(
+            $this->paymentStatusService->getPaymentStatusWithOrderId(
+                $request->get('orderId'),
```
Comment (cyattilakiss) [inline src/Controller/SalesChannelApiController.php:167]: please do a validation if the $request->get('orderId') is empty then return an error message

Comment (acampos1916) [reply src/Controller/SalesChannelApiController.php:172]: Done

File: src/Controller/SalesChannelApiController.php [hunk 6]
```diff
@@ -27,10 +27,12 @@
 
 use Adyen\Shopware\Service\PaymentDetailsService;
 use Adyen\Shopware\Service\PaymentMethodsService;
+use Adyen\Shopware\Service\PaymentStatusService;
+use http\Env\Response;
```
Comment (cyattilakiss) [inline src/Controller/SalesChannelApiController.php:31]: I don't think you will need this
```suggestion
```

Comment (acampos1916) [reply src/Controller/SalesChannelApiController.php:31]: Indeed, generated by the IDE at some point

**Pull request merged**